arduino/sketch/profiles.go

```diff
@@ -26,21 +26,23 @@ import (
 	"github.com/arduino/arduino-cli/arduino/utils"
 	"github.com/arduino/go-paths-helper"
 	semver "go.bug.st/relaxed-semver"
-	"gopkg.in/yaml.v2"
+	"gopkg.in/yaml.v3"
 )
 
 // Project represents the sketch project file
 type Project struct {
-	Profiles        Profiles `yaml:"profiles"`
-	DefaultProfile  string   `yaml:"default_profile"`
-	DefaultFqbn     string   `yaml:"default_fqbn"`
-	DefaultPort     string   `yaml:"default_port,omitempty"`
-	DefaultProtocol string   `yaml:"default_protocol,omitempty"`
+	ProfilesRaw     yaml.Node  `yaml:"profiles"`
+	Profiles        []*Profile `yaml:"-"`
```

Comment:
In the new yamlv3 package, they removed the `MapSlice` which would parse the map and respect the order of the key. 
So to achieve the same thing we have to use the `yaml.Node` type and do by hand what essentially the `MapSlice` was doing. The only downside is that we have to bring the `ProfilesRaw` just for parsing purposes. I thought that bringing a new struct for parsing purposes and copying the result back to the original struct was a bit too much in this case.

Reply:
> I thought that bringing a new struct for parsing purposes and copying the result back to the original struct was a bit too much in this case.

IMHO it's better to use a support structure, I don't like much this `ProfilesRaw` field, it's a publicly visible field that is confusing (should I look inside ProfilesRaw or Profiles? Should they be kept synchronized?). I see also that we cannot even make it private because we must unmarshal.
Another advantage is that the support structure may be discarded after being used.